futures-util/src/async_await/join.rs

```diff
@@ -1,5 +1,26 @@
 //! The `join` macro.
 
+/// Polls multiple futures simultaneously, returning a tuple
+/// of all results once complete. 
+/// 
+/// While `join!(a, b)` is similar to `(await!(a), await!(b))`,
+/// unlike the latter `join!` can make progress on multiple pending futures at once.
```

Comment:
That sentence sounds a bit weird. What do you think about this as replacement?:

``The difference between `join!(a, b)` and `(await!(a), await!(b))` is that `join!` polls both futures concurrently and therefore is more efficient.``